service/state/service.go

```diff
@@ -2,20 +2,29 @@ package state
 
 import (
 	"context"
+	"sync/atomic"
 
 	"github.com/celestiaorg/nmt/namespace"
+
+	"github.com/celestiaorg/celestia-node/fraud"
 )
 
 // Service can access state-related information via the given
 // Accessor.
 type Service struct {
 	accessor Accessor
+
+	fsub   fraud.Subscriber
+	cancel context.CancelFunc
+
+	befpReceived uint64
```

Comment:
You don't really need it here and we can simplify this by relying on the context. as we already have cancel there. Also, I don't see this being used anywhere. There is `BEFPReceived` but is unused.

Instead of having the uint64 here(which should have been uint32 btw), we can cancel the context and check if it was canceled before doing anything with
```go
if err := ctx.Err(); err != nil {
	return err
}
```

So if you want to prevent users from Submitting new txs - add the if above to `SubmitTx`.
Also:
* Remove s.cancel = nil from Stop, so there are no race conditions
* Move out `go fraud.OnBEFP` to DI

Reply:
Done